core/src/consensus.rs

```diff
@@ -130,6 +130,9 @@ pub const HARD_FORK_INTERVAL: u64 = YEAR_HEIGHT / 2;
 /// Floonet first hard fork height, set to happen around 2019-06-20
 pub const FLOONET_FIRST_HARD_FORK: u64 = 185_040;
 
+/// Floonet second hard fork height, set to happen around 2019-12-19
+pub const FLOONET_SECOND_HARD_FORK: u64 = 298_080;
+
 /// Check whether the block version is valid at a given height, implements
 /// 6 months interval scheduled hard forks for the first 2 years.
 pub fn valid_header_version(height: u64, version: HeaderVersion) -> bool {
```

Comment:
Not directly related to this PR but it would be _awesome_ if we had a function that took a block height and returned the valid header version. We could then refactor `valid_header_version()` to use this internally. We would _also_ be able to leverage it for block height specific logic elsewhere.

Reply:
Yes; I've also been thinking about adding that, just to simplify the logic in valid_header_version.
The latter does have the advantage in making certain heights invalid regardless of version, which is a nice way to stop obsolete clients in their track.

Reply:
see latest commit...